daft/table/table_io.py

```diff
@@ -263,6 +264,20 @@ def _to_file(
         )
         arrow_table = arrow_table.cast(downcasted_schema)
 
+    arrow_fields = []
```

Comment:
I think we might want to do this in Rust actually

I believe `Field.to_arrow()` might be the [correct place](https://github.com/Eventual-Inc/Daft/blob/f5837a2ecad409c72963ab4042de68c4238f1c54/src/daft-core/src/datatypes/field.rs#L20-L42).

We could even do something like ensuring that all `Field`s that get initialized at `Field::new()` with an image dtype will carry along the correct metadata.

cc @clarkzinzow for thoughts